cuqi/geometry/_geometry.py

```diff
@@ -581,22 +581,52 @@ def __eq__(self, obj):
 #         return self.mean + self.L.T@p
     
 class KLExpansion(Continuous1D):
-    '''
-    class representation of the random field in  the sine basis
-    alpha = sum_i p * (1/i)^decay * sin(i*L*x/pi)
-    L is the length of the interval
-    '''
+    """
+    Class representation of the random field in the sine basis
+
+    .. math::
+        f = \sum_{i=0}^{N-2} (\\frac{1}{(i+1)^\\text{decay}\\text{normalizer}})  p_i \\, sin(\\frac{\\pi}{N}(i+1)(K+\\frac{1}{2})) 
+        
+        + \\frac{(-1)^K}{2}(\\frac{1}{(N)^\\text{decay}\\text{normalizer}})  p_{N-1}
```

Comment:
Thank you @nabriis for your review and very helpful comments! The grid is not used in the `KLExpansion` code and I think this is because the discrete sine transform is used. I think this is still makes sense, the only thing here is I think the grid needs to be regular for this idea to make sense, that's why I added `regular` in the grid description (I can also possibly add a check and error if the grid is not).

Reply:
Ok. Sounds good to me.